containers/daap-python-base/tests/unit/curated_data/test_curated_data_loader.py

```diff
@@ -0,0 +1,163 @@
+import json
+import logging
+from unittest.mock import patch
+
+import pytest
+from curated_data.curated_data_loader import (
+    CuratedDataCopier,
+    format_parquet_glue_table_input,
+)
+from data_platform_paths import DataProductElement
+from data_product_metadata import DataProductSchema
+
+test_metadata = {
+    "name": "test_product",
+    "description": "just testing the metadata json validation/registration",
+    "domain": "MoJ",
+    "dataProductOwner": "matthew.laverty@justice.gov.uk",
+    "dataProductOwnerDisplayName": "matt laverty",
+    "email": "matthew.laverty@justice.gov.uk",
+    "status": "draft",
+    "retentionPeriod": 3000,
+    "dpiaRequired": False,
+    "schemas": ["test_table", "test_table1"],
+}
+
+
+changes1 = {
+    "removed_columns": ["col_1"],
+    "added_columns": None,
+    "types_changed": None,
+    "descriptions_changed": None,
+}
+
+changes2 = {
+    "removed_columns": ["col_1"],
+    "added_columns": None,
+    "types_changed": ["col_1"],
+    "descriptions_changed": None,
+}
+
+changes3 = {
+    "removed_columns": ["col_1"],
+    "added_columns": ["col_3", "col_4"],
+    "types_changed": None,
+    "descriptions_changed": ["col_2"],
+}
+
+valid_copy1 = True
+valid_copy2 = False
+valid_copy3 = True
+
+expected_copy1 = ["test_table", "test_table1"]
+expected_copy2 = ["test_table1"]
+expected_copy3 = ["test_table", "test_table1"]
+
+test_params1 = [
+    (changes1, valid_copy1, expected_copy1),
+    (changes2, valid_copy2, expected_copy2),
+    (changes3, valid_copy3, expected_copy3),
+]
+
+test_params2 = [
+    (changes1, expected_copy1),
+    (changes2, expected_copy2),
+    (changes3, expected_copy3),
+]
+
+
+class TestCuratedDataCopier:
+    @pytest.fixture(autouse=True)
+    def setup(self, metadata_bucket, s3_client):
+        self.s3_client = s3_client
+        self.bucket_name = metadata_bucket
+
+        s3_client.put_object(
+            Body=json.dumps(test_metadata),
+            Bucket=self.bucket_name,
+            Key=f"{test_metadata['name']}/v1.0/metadata.json",
+        )
+
+    @pytest.mark.parametrize("changes, valid_copy, expected_copy", test_params1)
+    def test_init(self, athena_client, glue_client, changes, valid_copy, expected_copy):
```

Comment:
[minor] I think you could skip this test, unless you consider `copy_updated_table` `table_to_copy` attributes to be an important part of the public interface of this class. The object will need to be initialised in every test, so the code will be exercised either way.